### PR TITLE
fix: Replace default values for option setting items with stack name as prefix

### PR DESCRIPTION
**Pull request by 96malhar**
*Description of changes:*
Currently in the deploy tool we are prefixing the default values for certain parameter (ECS Service name, Elastic Beanstalk Environment etc..) with the `ProjectName`. However, the `ProjectName` is not unique because we use the same project for multiple integration tests. This is causing an issue while deletion of stacks because the Cloudformation resources are shared between multiple stacks. 

In order to avoid this issue, we are replacing the `ProjectName` prefix is replaced with the `StackName`, which is unique per test. As a result, all Cloudformation resources created by a test are exclusive to the stack that it created

The same is also extended to the server mode.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by philasmar — APPROVED** [review]

File: src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs [hunk 1]
```diff
@@ -36,6 +36,8 @@ namespace AWS.Deploy.CLI.ServerMode.Controllers
     [Route("api/v1/[controller]")]
     public class DeploymentController : ControllerBase
     {
+        private const string REPLACE_TOKEN_STACK_NAME = "{StackName}";
```
Comment (normj) [inline src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs:39]: The constant should not be defined more then once.

File: src/AWS.Deploy.CLI/Commands/DeployCommand.cs [hunk 2]
```diff
@@ -25,6 +25,8 @@ namespace AWS.Deploy.CLI.Commands
 {
     public class DeployCommand
     {
+        private const string REPLACE_TOKEN_STACK_NAME = "{StackName}";
```
Comment (normj) [inline src/AWS.Deploy.CLI/Commands/DeployCommand.cs:28]: Why was this changed from `ProjectName` to `StackName`? I assume it is fine but these are the types of changes that would be breaking changes after we have users saving their deployment projects.

**Review by normj — CHANGES_REQUESTED** [review]
Lets us be better about PR descriptions. Explain what the root cause of the issue was and how this change fixes the issue.

File: src/AWS.Deploy.CLI/Commands/DeployCommand.cs (hunk 2 above)
Comment (96malhar) [reply src/AWS.Deploy.CLI/Commands/DeployCommand.cs:28]: Updated the PR description to provide clarity.

File: src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs (hunk 1 above)
Comment (96malhar) [reply src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs:39]: I have only defined this once in this class.

Comment (normj) [reply src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs:39]: It is also defined in `DeployCommand` which is the same project.

**Review by normj — CHANGES_REQUESTED** [review]

File: src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs (hunk 1 above)
Comment (96malhar) [reply src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs]: Fixed
